### PR TITLE
INTYGFV-15205: Implementation of new numeric component QuestionSjukskrivningsgrad. Added config CertificateDataConfigInteger & CertificateDataValueInteger.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**

### CODE REVIEW

File: skl/ag-parent/src/main/java/se/inera/intyg/common/agparent/model/converter/certificate/AbstractQuestionDiagnoser.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+                diagnosis.getCode(),
+                diagnosis.getTerminology(),
+                diagnosis.getDescription(),
+                moduleService.getDescriptionFromDiagnosKod(diagnosis.getCode(), diagnosis.getTerminology()));
+            newDiagnosisList.set(convertIdToInt(diagnosis.getId()), newDiagnosis);
+        });
+        return newDiagnosisList;
+    }
+
+    private static int getMaxDiagnosIndex(List<CertificateDataValueDiagnosis> diagnosisList) {
```
Comment (mhornfeldt) [inline skl/ag-parent/src/main/java/se/inera/intyg/common/agparent/model/converter/certificate/AbstractQuestionDiagnoser.java]: Should it be called getMaxDiagnosisIndex instead?

File: skl/ag-parent/src/main/java/se/inera/intyg/common/agparent/model/converter/certificate/AbstractQuestionDiagnoser.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+                    .build()
+            )
+            .validation(
+                new CertificateDataValidation[]{
+                    CertificateDataValidationMandatory.builder()
+                        .questionId(questionId)
+                        .expression(withCitation(singleExpression(DIAGNOSES_LIST_ITEM_1_ID)))
+                        .build(),
+                    CertificateDataValidationText.builder()
+                        .limit(LIMIT_DIAGNOSIS_DESC)
```
Comment (mhornfeldt) [inline skl/ag-parent/src/main/java/se/inera/intyg/common/agparent/model/converter/certificate/AbstractQuestionDiagnoser.java]: Does it not need id to be set?

File: skl/ag114/src/main/java/se/inera/intyg/common/ag114/v1/model/converter/certificate/question/QuestionSjukskrivningsgrad.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                    CertificateDataValidationMandatory.builder()
+                        .questionId(SJUKSKRIVNINGSGRAD_SVAR_ID)
+                        .expression(singleExpression(SJUKSKRIVNINGSGRAD_SVAR_JSON_ID))
+                        .build()
+                }
+            )
+            .build();
+    }
+
+    private static boolean isNumeric(String stringNumber) {
```
Comment (mhornfeldt) [inline skl/ag114/src/main/java/se/inera/intyg/common/ag114/v1/model/converter/certificate/question/QuestionSjukskrivningsgrad.java:74]: Maybe this function can be moved to a more general file. I saw that you added some functions to ValueToolkit below.

File: support/src/main/java/se/inera/intyg/common/support/facade/model/config/CertificateDataConfigInteger.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    @Getter(onMethod = @__(@Override))
+    String icon;
+    @Getter(onMethod = @__(@Override))
+    String text;
+    @Getter(onMethod = @__(@Override))
+    String description;
+    @Getter(onMethod = @__(@Override))
+    Accordion accordion;
+    String id;
+    String unit;
```
Comment (mww-sopra-steria) [inline support/src/main/java/se/inera/intyg/common/support/facade/model/config/CertificateDataConfigInteger.java:48]: Suggest that you call it unitOfMeasurement. Unit is a very generic term that means many things.

File: support/src/main/java/se/inera/intyg/common/support/facade/model/value/CertificateDataValueInteger.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+@JsonDeserialize(builder = CertificateDataValueIntegerBuilder.class)
+@Value
+@Builder
+public class CertificateDataValueInteger implements CertificateDataValue {
+
+    @Getter(onMethod = @__(@Override))
+    CertificateDataValueType type = CertificateDataValueType.INTEGER;
+    String id;
+    Integer value;
```
Comment (mww-sopra-steria) [inline support/src/main/java/se/inera/intyg/common/support/facade/model/value/CertificateDataValueInteger.java:36]: Suggest that we have unitOfMeasurement as part of the value. The value can be viewed and interpreted on its own.

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Nice! I would suggest a few minor changes to the model though.

File: support/src/main/java/se/inera/intyg/common/support/facade/model/config/CertificateDataConfigInteger.java (hunk 4 above)
Comment (ChrilleSigerhed) [reply support/src/main/java/se/inera/intyg/common/support/facade/model/config/CertificateDataConfigInteger.java:48]: Agreed, done.

File: support/src/main/java/se/inera/intyg/common/support/facade/model/value/CertificateDataValueInteger.java (hunk 5 above)
Comment (ChrilleSigerhed) [reply support/src/main/java/se/inera/intyg/common/support/facade/model/value/CertificateDataValueInteger.java:36]: Agreed, done.

File: skl/ag-parent/src/main/java/se/inera/intyg/common/agparent/model/converter/certificate/AbstractQuestionDiagnoser.java (hunk 2 above)
Comment (ChrilleSigerhed) [reply skl/ag-parent/src/main/java/se/inera/intyg/common/agparent/model/converter/certificate/AbstractQuestionDiagnoser.java]: Its not needed, but I still think we should set id. Done.

File: skl/ag-parent/src/main/java/se/inera/intyg/common/agparent/model/converter/certificate/AbstractQuestionDiagnoser.java (hunk 1 above)
Comment (ChrilleSigerhed) [reply skl/ag-parent/src/main/java/se/inera/intyg/common/agparent/model/converter/certificate/AbstractQuestionDiagnoser.java]: Good suggestion. Done.

File: skl/ag114/src/main/java/se/inera/intyg/common/ag114/v1/model/converter/certificate/question/QuestionSjukskrivningsgrad.java (hunk 3 above)
Comment (ChrilleSigerhed) [reply skl/ag114/src/main/java/se/inera/intyg/common/ag114/v1/model/converter/certificate/question/QuestionSjukskrivningsgrad.java:74]: Great point, moved it to ValueToolkit as suggested.

**Review by mhornfeldt — APPROVED** [review]

**Review by mww-sopra-steria — APPROVED** [review]
Looks great!

**Pull request merged**